src/game.py

```diff
@@ -0,0 +1,139 @@
+from src.client.game_client import Client
+from src.map.game_map import Map
+from src.players.player import Player
+from src.players.bot_player import BotPlayer
+
+from threading import Semaphore
+import pygame
+
+
+class Game:
+    def __init__(self, name: str = None, max_players: int = 1, num_turns: int = None) -> None:
+        self.__name: str = name
+        self.__map = None
```

Comment:
then use type annotations everywhere
 for example, you can use Optional[type] from typing